SIPS/sip-7.md

```diff
@@ -2,307 +2,205 @@
 sip: 7
 title: Snaps UI
 status: Draft
-discussions-to: https://github.com/MetaMask/SIPs/discussions/71
-author: Frederik Bolding (@FrederikBolding), Maarten Zuidhoorn (@Mrtenz)
+discussions-to: https://github.com/wallet/SIPs/discussions/71
+author: Frederik Bolding (@FrederikBolding), Maarten Zuidhoorn (@Mrtenz), Olaf Tomalka (@ritave)
 created: 2022-10-27
 ---
 
-## Abstract
+# Abstract
 
-This SIP proposes a way for Snaps to compose custom UIs using the existing implemented components of the MetaMask extension, while keeping security intact. The proposal specifies how a Snap can specify a UI using JSON, manage user input and update the UI on the fly. For this, this specification outlines a number of JSON-RPC methods for implementation.
+This SIP proposes a way for Snaps to compose custom UIs using the existing implemented components of the wallet extension, while keeping security intact. The proposal specifies how a Snap can specify a UI using JSON, manage user input and update the UI on the fly. For this, this specification outlines a number of JSON-RPC methods for implementation.
 
-This will allow for an improved user experience and customisability over the existing user interface solution for Snaps.
+This will allow for an improved user experience and better customization over the existing user interface solution for Snaps.
 
-## Motivation
+# Motivation
 
-One of the most requested features in Snaps is the ability to create a custom user interface. This should be done in such a way that it has the same look and feel as MetaMask, while still distinguishable from the main MetaMask user interface. By using a JSON-based format, Snaps can define their own user interface in a way that is safe to implement.
+One of the most requested features in Snaps is the ability to create a custom user interface. This should be done in such a way that it has the same look and feel as wallet, while still distinguishable from the main wallet user interface. By using a JSON-based format, Snaps can define their own user interface in a way that is safe to implement.
 
-## Specification
+# Specification
 
 > Formal specifications are written in TypeScript.
 
-### Language
+## Language
 
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",  "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119.](https://www.ietf.org/rfc/rfc2119.txt)
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119.](https://www.ietf.org/rfc/rfc2119.txt)
 
-### Definitions
+## Definitions
 
-```tsx
-type Json =
-  | null
-  | boolean
-  | number
-  | string
-  | Json[]
-  | { [prop: string]: Json };
```

Comment:
Why did you remove this? I think it's still important to clarify what we mean by "JSON-serializable".

Reply:
IMHO it shouldn't be part of this SIP to define what a JSON-serializable value is, because we used that term in multiple other places.